src/main/java/com/google/sps/servlets/AddOrganizationServlet.java

```diff
@@ -52,15 +60,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     openHours.add(Integer.parseInt(hourOpen));
     openHours.add(Integer.parseInt(hourClosed));
 
-    ArrayList<String> moderatorList = new ArrayList<String>();
-    moderatorList.add("anon creator"); //dummy value for before auth gets set up
+    /* This implementation stores history entries as embedded entities instead of custom objects
+     * because it is much simpler that way */
+    ArrayList changeHistory = new ArrayList<>();
+    UserService userService = UserServiceFactory.getUserService();
+    boolean isUserLoggedIn = userService.isUserLoggedIn();
+    String username;
+    if (isUserLoggedIn) {
+      /* Currently uses user email to be consistent w other parts of codebase, subject to change */
+      username = userService.getCurrentUser().getEmail();
+    } else {
+      username = "Unknown Author";
+    }
```

Comment:
Are anonymous users allowed to create organizations? How did we get here?

Reply:
No, this situation should never happen, I just added this so if it does happen or for whatever reason the userservice doesn't pick up an email it's flagged as an unknown author

Reply:
In general it'll be easier to make the service robust if you fail on conditions that shouldn't happen instead of accepting them. Either way you're being defensive, but by failing you:
 1) Get alerted to the fact that something is broken
 2) Stay more secure. (Eg a bug doesn't enable unauthorized users access to org creation)
 3) Write less code. (Preconditions are simpler than error handling)

Reply:
That makes sense- the new code throws an exception instead